### PR TITLE
HIVE-24591: Move Beeline To SLF4J Simple Logger

### PR DESCRIPTION
**Pull request by belugabehr**

### CODE REVIEW

File: bin/ext/cli.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ updateCli() {
     CLASS=org.apache.hadoop.hive.cli.CliDriver
     JAR=hive-cli-*.jar
   else
-    export HADOOP_CLIENT_OPTS=" -Dproc_beeline $HADOOP_CLIENT_OPTS -Dlog4j.configurationFile=beeline-log4j2.properties"
+    export HADOOP_CLIENT_OPTS="-Dproc_beeline -Dorg.slf4j.simpleLogger.defaultLogLevel=warn $HADOOP_CLIENT_OPTS"
```
Comment (nrg4878) [inline bin/ext/cli.sh:31]: dumb question? how would users enable debug logging in beeline with this approach ?

File: beeline/pom.xml [hunk 2]
```diff
@@ -105,19 +117,40 @@
       <artifactId>super-csv</artifactId>
       <version>${super-csv.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
```
Comment (nrg4878) [inline beeline/pom.xml:122]: arent we already using slf4j libraries in beeline today? How has this been compiling without explicit dependencies? transitive?
also we should specify a scope for these dependencies.

**Comment by nrg4878** [comment]
@belugabehr Could you add a bit more details on the rationale behind these changes in the jira, so I/we can understand what user facing impact, this fix has. I am not sure I understand what is broken or needs to be enhanced. It is my impression beeline already uses slf4j.

File: bin/ext/cli.sh (hunk 1 above)
Comment (belugabehr) [reply bin/ext/cli.sh:31]: Not a dumb question.  I think I included it in the original JIRA, but here it is:

```
export HADOOP_CLIENT_OPTS="-Dorg.slf4j.simpleLogger.defaultLogLevel=debug"
beeline ...
```

File: beeline/pom.xml (hunk 2 above)
Comment (belugabehr) [reply beeline/pom.xml:122]: It is being used, but the dependencies have been transitive for a long time (d'oh).

No scope required. Technically don't need it for compile, but it's no harm.

**Comment by belugabehr** [comment]
@nrg4878 beeline uses Log4J as the backend logger.  Log4J is a pretty robust, heavy, logging implementation that plugs into SLF4J.  I have proposed here that we swap out Log4J and use the 'Simple' logger that is produced by SLF4J team.  This will make configuring beeline debug logging much easier.  Sure, we won't be able to forward beeline log statements to JMS and Kafka anymore, but who would want to do that anyway? :)

**Comment by belugabehr** [comment]
Not to self, make scope for logger `runtime`

**Comment by belugabehr** [comment]
Not to self, need to make sure that only one log implementation is on the class path, this may be tricky

**Review by nrg4878 — APPROVED** [review]

**Comment by nrg4878** [comment]
Changes look good to me. +1

**Review by achennagiri — APPROVED** [review]
Looks good to me!

**Comment by belugabehr** [comment]
Eh. It actually needs some work.

Since HiveServer and Beeline share a lib directory (if they are co-installed), I need to update the start scripts to deal with that since HS2 uses log4j and beeline uses simple logger.... there will be issues if they are on the same classpath at the same time. That's why it will be nice if beeline is isolated to its own distribution. @nrg4878

**Comment by achennagiri** [comment]
@belugabehr https://github.com/apache/hive/pull/1906 This PR isolates beeline into it's own distribution but the bin distribution is also going to have the beeline script. Should I remove the beeline scripts completely from there and just keep it in the new distribution?

**Pull request closed without merging**